repos/system_upgrade/common/libraries/module.py

```diff
@@ -0,0 +1,71 @@
+import warnings
+
+try:
+    import dnf
+except ImportError:
+    dnf = None
+    warnings.warn('Could not import the `dnf` python module.', ImportWarning)
+
+try:
+    import hawkey
+except ImportError:
+    hawkey = None
+    warnings.warn('Could not import the `hawkey` python module.', ImportWarning)
+
+
+def _create_or_get_dnf_base(base=None):
+    if not base:
+        base = dnf.Base()
+        base.read_all_repos()
+        base.fill_sack()
+    return base
+
+
+def get_modules(base=None):
+    """
+    Return info about all module streams as a list of libdnf.module.ModulePackage objects.
+    """
+    if not dnf:
+        return []
+    base = _create_or_get_dnf_base(base)
+
+    module_base = dnf.module.module_base.ModuleBase(base)
+    # this method is absent on RHEL 7, in which case there are no modules anyway
+    if not hasattr(module_base, 'get_modules'):
+        return []
+    return module_base.get_modules('*')[0]
+
+
+def get_enabled_modules():
+    """
+    Return currently enabled module streams as a list of libdnf.module.ModulePackage objects.
+    """
+    if not dnf:
+        return []
+
+    base = _create_or_get_dnf_base()
+    modules = get_modules(base)
+
+    # if modules are not supported (RHEL 7), base.sack._moduleContainer won't exist
+    # luckily in that case modules are empty and the element won't even be accessed
+    return [m for m in modules if base.sack._moduleContainer.isEnabled(m)]
```

Comment:
because it's not about consistency but how it is used. get_enabled_modules() just produces it, get_modules is used elsewhere too where the base was already created. get_enabled_modules isn't.